internal/app/mistapiconnector/mistapiconnector_app.go

```diff
@@ -551,15 +558,15 @@ func (mc *mac) handleDefaultStreamTrigger(w http.ResponseWriter, r *http.Request
 }
 
 // putEtcdKeys puts keys in one transaction
-func (mc *mac) putEtcdKeys(playbackID string, kvs ...string) error {
+func (mc *mac) putEtcdKeys(sess *concurrency.Session, playbackID string, kvs ...string) error {
 	if len(kvs) == 0 || len(kvs)%2 != 0 {
 		return errors.New("number of arguments should be even")
 	}
 	cmp := clientv3.Compare(clientv3.CreateRevision(kvs[0]), ">", -1) // basically noop - will always be true
 	thn := make([]clientv3.Op, 0, len(kvs)/2)
 	get := clientv3.OpGet(kvs[0])
 	for i := 0; i < len(kvs); i += 2 {
-		thn = append(thn, clientv3.OpPut(kvs[i], kvs[i+1], clientv3.WithLease(mc.etcdSession.Lease())))
+		thn = append(thn, clientv3.OpPut(kvs[i], kvs[i+1], clientv3.WithLease(sess.Lease())))
```

Comment:
I am wondering if, at least when "recovering" the keys, we should have a different `clientv3.Compare` function here, like to only set the `etcd` keys iff the current key mod (create?) revision is `0` (doesn't exist?) or exactly what the revision we had before (to put it simply: `modRev < myRev+1`).

Is it possible that when one `mist-api-connector` instance loses the session/lease, the stream would immediately go to some `mist-api-connector` instance which would create different routes for it?

Reply:
> Is it possible that when one mist-api-connector instance loses the session/lease, the stream would immediately go to some mist-api-connector instance which would create different routes for it?

When mist-api-connector looses session it doesn't mean that connection stopped. When connection stopped Mist sends CLOSE_CONN trigger to mist-api-connector. So only case when connection is stopped but mist-api-connector is not aware about that - is when Mist crashes, which hasn't happened for us for a long time.